kernel/thread.c

```diff
@@ -468,24 +466,18 @@ Z_SYSCALL_HANDLER(k_thread_create,
 						_OBJ_INIT_FALSE) == 0,
 				    "bad stack object"));
 
-#ifndef CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT
 	/* Verify that the stack size passed in is OK by computing the total
 	 * size and comparing it with the size value in the object metadata
-	 *
-	 * We skip this check for SoCs which utilize MPUs with power of two
-	 * alignment requirements as the guard is allocated out of the stack
-	 * size and not allocated in addition to the stack size
 	 */
-	guard_size = (u32_t)K_THREAD_STACK_BUFFER(stack) - (u32_t)stack;
-	Z_OOPS(Z_SYSCALL_VERIFY_MSG(!__builtin_uadd_overflow(guard_size,
+	Z_OOPS(Z_SYSCALL_VERIFY_MSG(!__builtin_uadd_overflow(K_THREAD_STACK_RESERVED,
 							     stack_size,
 							     &total_size),
 				    "stack size overflow (%u+%u)", stack_size,
-				    guard_size));
-#else
-	total_size = stack_size;
-#endif
-	/* They really ought to be equal, make this more strict? */
+				    K_THREAD_STACK_RESERVED));
+
+	/* Testing less-than-or-equal since additional room may have been
```

Comment:
We wouldn't have to do this @andrewboie if we were forced (by API) to always pass K_THREAD_STACK_SIZEOF(K_THREAD_STACK_DEFINE(sym)) to k_thread_stack_create.